egs2/TEMPLATE/asr1/asr.sh

```diff
@@ -510,6 +510,11 @@ if ! "${skip_data_prep}"; then
                 <data/"${dset}"/cmvn.scp awk ' { print($1,"<DUMMY>") }' > data/"${dset}"/wav.scp
                 utils/copy_data_dir.sh --validate_opts --non-print data/"${dset}" "${data_feats}${_suf}/${dset}"
 
+                # Derive the the frame length and feature dimension
+                _nj=$(min "${nj}" "$(<"${data_feats}${_suf}/${dset}/utt2spk" wc -l)")
+                scripts/feats/feat_to_shape.sh --nj "${_nj}" --cmd "${train_cmd}" \
+                    "${data_feats}${_suf}/${dset}/feats.scp" "${data_feats}${_suf}/${dset}/feats_shape"
```

Comment:
Where is `feats_shape` used?
Is it needed?

Reply:
Yes, it is needed by [here](https://github.com/espnet/espnet/blob/06335e6a427e174d1db5e68c97fba944451426f5/egs2/TEMPLATE/asr1/asr.sh#L571) .
In that `else` branch, except `raw` , the other types of freature require `feats_shape`.